### PR TITLE
Improve upload speed

### PR DESCRIPTION
**Pull request by AaronH88**
Created a pool of workers to upload files and improve the speed of uploading files. Jira ticket THEEDGE-973

Everything works locally, will need to calibrate the number of workers in stage.

### CODE REVIEW

File: pkg/services/files/uploader.go [hunk 1]
```diff
@@ -89,6 +90,24 @@ func newS3Uploader() *S3Uploader {
 	}
 }
 
+type filePathVar struct {
```
Comment (biancarosa) [inline pkg/services/files/uploader.go:93]: Thinking about code readability, can we rename this? I thought about repoSettings, localRepo, repoToUpload, a bunch of things that I'm not particularly happy either but at least it's clearer 😬  @maxamillion @acosferreira @loadtheaccumulator if you have better ideas that's the time

File: pkg/services/files/uploader.go [hunk 2]
```diff
@@ -89,6 +90,24 @@ func newS3Uploader() *S3Uploader {
...
+	uploadPath string
+}
+
+func worker(wg *sync.WaitGroup, queue <-chan filePathVar, u *S3Uploader) {
+	defer wg.Done()
+
+	for p := range queue {
+		fname := p.fileName
+		uploadDest := p.uploadPath
+		_, err := u.UploadFile(fname, uploadDest)
```
Comment (biancarosa) [inline pkg/services/files/uploader.go:104]:
```suggestion
		_, err := u.UploadFile(p.filename, p.uploadDest)
```

Or just pass p right away, but that's a major refactor

File: pkg/services/files/uploader.go [hunk 3]
```diff
@@ -97,10 +116,17 @@ func (u *S3Uploader) UploadRepo(src string, account string) (string, error) {
...
 	log.Debugf("S3Uploader::UploadRepo::account: %#v", account)
 
-	// FIXME: might experiment with doing this concurrently but I've read that
-	//		  that can get you rate limited by S3 pretty quickly so we'll mess
-	//		  with that later.
+	//queue of jobs
+	q := make(chan filePathVar)
+
+	numberOfWorkers := 100
+	var workWg sync.WaitGroup
```
Comment (biancarosa) [inline pkg/services/files/uploader.go:123]: workWg could also be something easier to read 😬 is this wait group used for files?

**Review by biancarosa — CHANGES_REQUESTED** [review]
@AaronH88 awesome work, minor comments, other than that we're good!

File: pkg/services/files/uploader.go (hunk 1 above)
Comment (AaronH88) [reply pkg/services/files/uploader.go:93]: How about "uploadDetails" ?

File: pkg/services/files/uploader.go (hunk 3 above)
Comment (AaronH88) [reply pkg/services/files/uploader.go:123]: Yeah, and its a unique group for the provided repo,
perhaps "repoWaitGroup" ?

File: pkg/services/files/uploader.go [hunk 4]
```diff
@@ -89,6 +95,33 @@ func newS3Uploader() *S3Uploader {
...
 }
 
+type uploadDetails struct {
+	fileName   string
+	uploadPath string
+	uploader   *S3Uploader
+	wg         *sync.WaitGroup
+}
+
+func worker() {
```
Comment (AaronH88) [inline pkg/services/files/uploader.go:105]: Rename to uploadWorkers() ??

**Review by biancarosa — APPROVED** [review]
Most of my comments were regarding readability and I feel like they've been addressed, I'm good with this and I think it'll be an amazing improvement!

File: pkg/services/files/uploader.go (hunk 4 above)
Comment (biancarosa) [reply pkg/services/files/uploader.go:105]: nah, in this case maybe we don't need to worry too much, it's pretty clear to me

File: pkg/services/files/uploader.go (hunk 3 above)
Comment (AaronH88) [reply pkg/services/files/uploader.go]: Changed to repoWaitGroup

File: pkg/services/files/uploader.go (hunk 2 above)
Comment (AaronH88) [reply pkg/services/files/uploader.go]: I agree that P is a major refactor, however I changed this fucntion call to use the "p." notation you suggested

**Pull request merged**